### PR TITLE
fix rustdoc generic param order

### LINKED ISSUE #61292: Send/Sync type parameters come after const parameters in docs
https://github.com/rust-lang/rust/blob/721268583759224d0f6476e0b8b196cc8afbdea0/src/test/rustdoc/generic-const.rs#L12-L13
Const parameters should always be trailing (at least with respect to type/lifetime parameters). I'm not sure whether this is specifically a rustdoc issue or not yet.

### PR DESCRIPTION
**Pull request by lcnr**
fixes #61292

r? @varkor cc @GuillaumeGomez

### CODE REVIEW

File: src/librustdoc/clean/simplify.rs [hunk 1]
```diff
@@ -118,18 +117,6 @@ pub fn merge_bounds(
...
 
-pub fn ty_params(mut params: Vec<clean::GenericParamDef>) -> Vec<clean::GenericParamDef> {
-    for param in &mut params {
-        match param.kind {
-            clean::GenericParamDefKind::Type { ref mut bounds, .. } => {
-                *bounds = mem::take(bounds);
-            }
-            _ => panic!("expected only type parameters"),
-        }
-    }
```
Comment (lcnr) [inline src/librustdoc/clean/simplify.rs:129]: Am I missing something here? What

**Comment by GuillaumeGomez** [comment]
Looks good to me! It even simplifies the code so that's great. 👍 

@bors: r+

**Comment by Manishearth** [comment]
Hmm, surprised that CI builds are taking three hours again

**Comment by lcnr** [comment]
Let's see if this was spurious I guess

@bors r=GuillaumeGomez

**Pull request merged**